### PR TITLE
Finally Terra Helm crafting closing #766

### PR DESCRIPTION
**Pull request by Lazersmoke**
I added a crafting handler for Helm's of revealing that copy the NBT from the input helm to the output helm. I am only 40% sure that this will not completely break something, but if it does then blame [my cat](http://popcorn.no-ip.info/Ninja%20Flail.gif). The class was derived from the Terra Shatterer Tipping recipe FYI. #766

### CODE REVIEW

File: src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+			ItemStack stack = var1.getStackInSlot(i);
+			if(stack != null && checkIfHelm(stack))
+				Helm = stack;
+		}
+
+		if(Helm == null)
+			return null;
+
+		ItemStack helmCopy = Helm.copy();
+		switch (checkHelm(helmCopy)):
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java]: No space after switch.

File: src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+			case 1:	ItemStack newHelm = new ItemManasteelHelmRevealing(helmCopy);
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java]: Newline after case.

File: src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java [hunk 3]
```diff
@@ -0,0 +1,91 @@
...
+	public int getRecipeSize() {
+		return 10; //WTF does this do?
+	}
+
+	@Override
+	public ItemStack getRecipeOutput() {
+		return null;
+	}
+	
+	private int checkHelm(ItemStack helmStack) { //Tried to do a switch/case but got too complex
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java]: Did you test this? ItemTerrasteelHelm : ItemManastellHelm, it'll always return 1.

File: src/main/java/vazkii/botania/common/item/interaction/thaumcraft/ItemElementiumHelmRevealing.java [hunk 4]
```diff
@@ -28,6 +28,14 @@ public ItemElementiumHelmRevealing() {
 		super(LibItemNames.ELEMENTIUM_HELM_R);
 	}
 
+	public ItemElementiumHelmRevealing(ItemElementiumHelm oldHelm) { //Used when crafting
+
```
Comment (Vazkii) [inline src/main/java/vazkii/botania/common/item/interaction/thaumcraft/ItemElementiumHelmRevealing.java]: Unpad this.

**Comment by Vazkii** [comment]
Ok so I was trying to look at the code to figure out wtf you were doing. Did you code this inside github? Because that's the only reasonable explanation as this won't even compile.

File: src/main/java/vazkii/botania/common/item/interaction/thaumcraft/ItemElementiumHelmRevealing.java (hunk 4 above)
Comment (nabijaczleweli) [inline src/main/java/vazkii/botania/common/item/interaction/thaumcraft/ItemElementiumHelmRevealing.java]: Such `camelCase` much wow

File: src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java (hunk 1 above)
Comment (nabijaczleweli) [inline src/main/java/vazkii/botania/common/crafting/recipe/HelmRevealingRecipe.java]: IIRC an opening bracket after a switch-statement is needed.

**Comment by Lazersmoke** [comment]
*retires to bug report duty*

**Pull request closed without merging**